### PR TITLE
Replace PhantomJS with XVFB run Chromium browser

### PR DESCRIPTION
**Pull request by hatched**
To support writing ES6 code without heavy transpiling we can no longer use PhantomJS as it's quite out of date. This branch removes PhantomJS from the test runner and replaces it with XVFB run Chromium.

### CODE REVIEW

**Comment by hatched** [comment]
#### No QA

If the tests pass we're good!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5553/
Test PASSed.

File: scripts/test-js-old.sh [hunk 1]
```diff
@@ -3,26 +3,32 @@
...
+cp karma-mocha-old.conf.js.tmpl karma-mocha-old.conf.js
+sed -i -e 's/{TEST_PORT}/'$PORT'/' karma-mocha-old.conf.js
+# Start Xvfb to run chromium in for the tests.
+Xvfb :99 & echo $! > $XVFB_PID
+# Start the asset server.
 bin/pserve test.ini test_port=$PORT & echo $! > $SERVE_PID
 
 finished () {
+  echo "Shutting down Xvfb."
+  kill -9 `cat xvfb_pid`
```
Comment (huwshimi) [inline scripts/test-js-old.sh]: Could you use `XVFB_PID` here?

File: scripts/test-js.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+set -m
+
+XVFB_PID="xvfb_pid"
+
+# Start the Xvfb to run chromium in for the tests.
+Xvfb :99 & echo $! > $XVFB_PID
+
+finished () {
+  echo "Shutting down Xvfb."
+  kill -9 `cat xvfb_pid`
```
Comment (huwshimi) [inline scripts/test-js.sh]: As above, could you use `XVFB_PID` here?

**Comment by huwshimi** [comment]
QA OK :+1: with a couple of questions.

File: scripts/test-js-old.sh (hunk 1 above)
Comment (hatched) [inline scripts/test-js-old.sh]: Good idea, updated

File: scripts/test-js.sh (hunk 2 above)
Comment (hatched) [inline scripts/test-js.sh]: Good idea, updated

**Comment by hatched** [comment]
Thanks @huwshimi, updates made!

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5555/
Test PASSed.

**Pull request merged**